chemclipse/plugins/org.eclipse.chemclipse.support/src/org/eclipse/chemclipse/support/settings/parser/SettingsClassParser.java

```diff
@@ -119,7 +120,7 @@ public SettingsClassParser(Class<?> settingclass) {
 							} else if(annotation instanceof DoubleSettingsProperty) {
 								DoubleSettingsProperty settingsProperty = (DoubleSettingsProperty)annotation;
 								inputValue.addValidator(new MinMaxValidator<Double>(property.getName(), settingsProperty.minValue(), settingsProperty.maxValue(), Double.class));
-							} else if(annotation instanceof StringSettingsProperty) {
+							} else if(annotation instanceof StringSettingsProperty && ((StringSettingsProperty)annotation).regExp() != null && !((StringSettingsProperty)annotation).regExp().isEmpty()) {
```

Comment:
This check should be placed inside the if to make it more readable.

Reply:
Resolved with a31a1d3692fac925e7e38d76f9515c91dd84d323.

Reply:
resolved with a31a1d3692fac925e7e38d76f9515c91dd84d323.